### ISSUE TITLE
Fix windows 11 defaults: enable UEFI, add TPM

### ISSUE THREAD

**Issue by berrange** [virt-manager/virt-manager#310]
I provisioned a new VM on Fedora 34 host using Windows 11 DVD ISO images.

I copied the win-11.xml def for libosinfo into /etc/osinfo/os/microsoft.com

https://gitlab.com/libosinfo/osinfo-db/-/merge_requests/362

and went through the new VM wizard, but configuration it creates is not suitable for Windows 11. The two key problems are

 - Use of EFI firmware is mandatory. Legacy BIOS is discontinued.  The win-11.xml for libosinfo declares support for EFI. I manually added firmware='efi' to the <os> before starting the installation at end of the wizard
 - Use of TPM devices is mandatory. I used "Add hardware" to add a TPM with default settings to the VM before starting the installation

With the two tweaks above, Windows 11 installer was able to complete.

There is a further issue with graphics, causing it to be limited to 800x600 regardless of which graphics card I pick. 

I think it might make sense to use 'EFI' firmware as a indicator that we should add a TPM by default too, as I suspect all modern OS that care about having TPMs will support EFI too.  With this in mind, we might not need to track TPM support against OS directly, especially as having a TPM is harmless if the guest doesn't need it.

One minor annoyance, swtpm_setup is broken when running with qemu:///session unless the user manually invokes '/usr/share/swtpm/swtpm-create-user-config-files' first, as a one time only task.

**Comment by sej7278** [comment]
can't really do that until swtpm and libtpms are packaged as a dependency of virt-manager

**Comment by sej7278** [comment]
yes the regedit hacks are just that - hacks, and i think they only work on an upgrade not a fresh install from iso.

as for dependencies, ok sure they could be dependencies of libvirt, but my point being you can't expect virt-manager to enable tpm by default until we get to the point where the tpm emulator libs are guaranteed to be installed.

**Comment by mhussaincov93** [comment]
hi,
i'm blind and booting from a windows 11 iso in efi mode,
no audio is well, audable.
gennerally, I use windows narrator when the windows setup is booted e.g when it asks for your language to install and keybord layout etc.
this is in bios mode, for win10
am i missing something?
is there a time out when booting via efi?
if yes could this timeout be made configurable?
to give me time to press enter or what ever to boot the iso?

stepps to repro.
1.
get a win11 iso of your choise.
2.
create a win11 vm in virt-manager setting it to boot via efi.
3.
press windows plus control plus enter to launch narrator.
what should happen,
I should hear narrator speaking to me.
what I hear,
nuthing.
thanks for your assistence in advanse,
commented here because I was requested to.
Majid

**Comment by martinpitt** [comment]
Just to make sure I understood it correctly: PR #341 provided the EFI bit, but the TPM half of this is still an issue, hence this is still open. But it wasn't rejected (I think it's the right thing to do), just postponed until https://gitlab.com/libosinfo/osinfo-db/-/merge_requests/362 gets settled? Thanks!

**Comment by crobinso** [comment]
@martinpitt If using virt-install and user selects UEFI and libvirt+qemu advertise TPM support, virt-install defaults to adding a TPM now, independent of Windows 11. The referenced osinfo-db PR is adding a `win11` entry: this still isn't applied yet. Once it's applied, current virt-install should do the right thing with UEFI and TPM default. So on our side this issue is mostly solved, but I'll keep it open until win11 lands in osinfo-db

**Comment by victortoso** [comment]
> but I'll keep it open until win11 lands in osinfo-db

Should be able to close now

**Comment by crobinso** [comment]
> > but I'll keep it open until win11 lands in osinfo-db
> 
> Should be able to close now

Thanks for the pointer. I also added some additional unit testing for virt-install + win11
